packages/components/src/components/Select/styles/select.module.scss

```diff
@@ -0,0 +1,216 @@
+        color: var(--text-color-x-weak);
+    }
+
+    &:has(input:placeholder-shown) {
+        color: var(--text-color-x-weak);
+    }
+}
+
+.input {
+    width: 100%;
+    background-color: transparent;
+    display: flex;
+    align-items: center;
+    text-align: inherit;
+    text-indent: sizeToken(3);
+    outline: 2px solid transparent;
+    outline-offset: 2px;
+    border-radius: calc(var(--radius) - var(--line-width));
+
+    &::placeholder {
+        color: var(--text-color-x-weak);
+    }
+
+    // Remove border-radius and text-indent/padding on the left if there’s a left-side slot
+    &:has(~ div[data-name='left']) {
+        text-indent: 0px;
+        border-top-left-radius: 0px;
+        border-bottom-left-radius: 0px;
+    }
+
+    // Remove border-radius and padding on the right if there’s a right-side slot
+    &:has(~ .slot[data-side='right'], ~ .slot:not([data-side='right'])) {
+        padding-right: 0;
+        border-top-right-radius: 0;
+        border-bottom-right-radius: 0;
+    }
+
+    &:disabled {
+        color: var(--text-color-disabled);
+    }
+    &:disabled ~ .slot {
+        color: var(--text-color-disabled);
+    }
+}
+
+.slot {
+    display: flex;
+    flex-shrink: 0;
+    align-items: center;
+    cursor: text;
+    gap: sizeToken(1.5);
+    padding-left: sizeToken(3);
+    padding-right: sizeToken(2);
+    color: var(--text-color);
+
+    // Left slot
+    &[data-name='left'] {
+        order: -1;
+        margin-left: 0px;
+        margin-right: 0px;
+    }
+
+    // Right slot
+    &[data-name='right'] {
+        order: 0;
+        margin-left: 0px;
+        margin-right: 0px;
+        padding-right: sizeToken(3);
+
+        &:has(~ div[data-name='clear']) {
+            padding-right: 0;
+        }
+    }
+
+    //Label slot
+    &[data-name='label'] {
+        display: flex;
+        align-items: center;
+        cursor: text;
+        height: 100%;
+        padding-left: sizeToken(3);
+        padding-right: sizeToken(2);
+
+        &:has(~ div[data-name='left']) {
+            text-indent: 0px;
+            padding-left: 0;
+        }
+    }
+
+
+}
+
+.clear {
+    display: flex;
+    height: 100%;
+    width: fit-content;
+    align-items: center;
+    justify-content: center;
+    padding: 0 sizeToken(3);
+    color: var(--text-color-weak);
+
+    &:hover {
+        color: var(--text-color-weak-hover);
+    }
+
+}
+
+.caret {
+    display: flex;
+    height: 100%;
+    width: sizeToken(4);
+    align-items: center;
+    justify-content: center;
+    margin-right: sizeToken(3);
+    color: var(--text-color-weak);
+
+
+
+    &:hover {
+        color: var(--text-color-weak-hover);
+    }
+}
+
+.menu {
+    display: none;
+    border-radius: var(--radius);
+    margin-top: sizeToken(2);
+    background-color: var(--base-color);
+    border: var(--line-width) solid var(--line-color-strong);
+    width: var(--radix-popover-trigger-width);
+    overflow: hidden;
+
+    &:has(li) {
```

Comment:
Instead of having global `li` styling, could we use data attr?

Reply:
i added the role to the selector so it only selects `li` with `role="option"`